### PR TITLE
NNBD linting

### PR DESCRIPTION
**Pull request by pq**
Cheers for these awesome lints @a14n!

Follow-up from https://github.com/dart-lang/linter/pull/2457.

/cc @bwilkerson

### CODE REVIEW

File: test/integration_test.dart [hunk 1]
```diff
@@ -128,6 +128,7 @@ void coreTests() {
 
         final options = _getOptionsFromString(src);
         var configuredLints =
+            // ignore: cast_nullable_to_non_nullable
             (options['linter'] as YamlMap)['rules'] as YamlList;
```
Comment (bwilkerson) [inline test/integration_test.dart:132]: This seems like a false positive to me. IIRC, `options` is a `Map<String, dynamic>`. While it's true that `dynamic` is nullable, I think it's reasonable to cast it to either a nullable or non-nullable type.

**Review by bwilkerson — APPROVED** [review]

**Comment by bwilkerson** [comment]
Only one of those lints has a quick fix associated with it. Out of curiosity, did you notice any patterns while you were fixing them that would be reasonable to offer as a quick fix?

**Comment by pq** [comment]
> Only one of those lints has a quick fix associated with it. Out of curiosity, did you notice any patterns while you were fixing them that would be reasonable to offer as a quick fix?

Removing an unnecessary `!` seems like an easy one and would be handy as a bulk fix.  The others are trickier.

File: test/integration_test.dart (hunk 1 above)
Comment (pq) [reply test/integration_test.dart:132]: fyi @a14n

**Pull request merged**